lib/tasks/delete_analytics_event_via_assessment_code.rake

```diff
@@ -0,0 +1,37 @@
+namespace :migrate do
+  desc "EL-2268: Delete unwanted data from AnalyticEvent table based on assessment code"
+  # Rake task to delete unwanted analytics event data.
+  # Call with "rake migrate:delete_analytics_events_based_on_assessment_code[mock]"
+  # e.g. rake "migrate:delete_analytics_events_based_on_assessment_code[false]"
+  # Running with mock=true will output the number of records to be deleted without actually deleting any
+  # (to allow for testing).
+
+  task :delete_analytics_events_based_on_assessment_code, %i[mock] => :environment do |_task, args|
+    if args.count != 1
+      Rails.logger.info 'call with rake "migrate:delete_analytics_events_based_on_assessment_code[mock]"'
+      next
+    end
+
+    mock = args[:mock].to_s.downcase.strip != "false"
```

Comment:
I find the use of `mock` to be a little confusing. 

In my mind, if mock is true then it is a mock so we would not expect it to delete but instead do a dry-run.

Everything else looks ok to me and as it is just a naming thing then it is not a blocker, but there may be a better name for this variable that is less confusing.